src/tools/rados/rados.cc

```diff
@@ -3120,15 +3120,21 @@ static int rados_tool_common(const std::map < std::string, std::string > &opts,
     cerr << "WARNING: pool copy does not preserve user_version, which some "
 	 << "apps may rely on." << std::endl;
 
-    if (rados.get_pool_is_selfmanaged_snaps_mode(src_pool)) {
+    ret = rados.pool_is_in_selfmanaged_snaps_mode(src_pool);
+    if (ret < 0) {
+      cerr << "failed to query pool " << src_pool << " for selfmanaged snaps: "
+           << cpp_strerror(ret) << std::endl;
+      return 1;
+    }
+    if (ret > 0) {
```

Comment:
nit: after seeing the initial `if(ret < 0)` I started doubke0-checking whether the block above doesn't modify `ret`. How about using `else` instead or making `ret` a constant?

Reply:
> How about using `else` instead

That would be an "else after return", which is thrown upon by many.

> or making `ret` a constant?

`ret` is declared ~1200 lines above ;)

Reply:
> That would be an "else after return", which is thrown upon by many.

Yes but right now we have:

```cpp
if (ret < 0) {
  return;
}
if (ret > 0) {
  // ...
}
```

which, from the point of indentation, has all the disadvantages of `else`  but additionally it draws extra attention. How about getting rid of it?

Reply:
> How about getting rid of it?

You said "yes" to the style concern, so I'm not sure what you mean by "get rid of it".  Is
```
if (ret < 0) {
  // ...
  return;
} else if (ret > 0) {
  // ...
}
```
what you would rather see?

Reply:
Yes but it's still a nit, so I'm not going to fight for it :-). The intention behind is to somehow tell a reader these are 2 checks for the same call / assignment. IMHO 2 separated `if` suggest an extra assignment happens in the first check.

Reply:
Obliged ;)